src/components/application_manager/test/commands/mobile/delete_file_test.cc

```diff
@@ -0,0 +1,146 @@
+#include "application_manager/smart_object_keys.h"
+#include "application_manager/commands/commands_test.h"
+#include "application_manager/commands/command_request_test.h"
+#include "application_manager/application.h"
+#include "application_manager/mock_application_manager.h"
+#include "application_manager/mock_application.h"
+#include "application_manager/mock_message_helper.h"
+#include "application_manager/event_engine/event.h"
+#include "application_manager/mock_hmi_capabilities.h"
+#include "application_manager/policies/mock_policy_handler_interface.h"
+#include "application_manager/mock_application_manager_settings.h"
+
+#include "interfaces/MOBILE_API.h"
+#include "application_manager/policies/policy_handler_interface.h"
+#include "application_manager/policies/policy_handler.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace delete_file {
+
+using ::testing::_;
+using ::testing::Test;
+using ::testing::Return;
+using ::testing::ReturnRef;
+namespace am = ::application_manager;
+using am::commands::DeleteFileRequest;
+using am::commands::DeleteFileResponse;
+using am::commands::MessageSharedPtr;
+using am::event_engine::Event;
+using am::MockMessageHelper;
+
+typedef SharedPtr<DeleteFileRequest> DeleteFileRequestPtr;
+typedef SharedPtr<DeleteFileResponse> DeleteFileResponsePtr;
+typedef NiceMock<
+    ::test::components::application_manager_test::MockHMICapabilities>
+    MockHMICapabilities;
+
+ACTION_TEMPLATE(SetArgPointer,
+                HAS_1_TEMPLATE_PARAMS(int, k),
+                AND_1_VALUE_PARAMS(vec)) {
+  *std::tr1::get<k>(args) = *vec;
+}
+
+namespace {
+const uint32_t kConnectionKey = 2u;
+const uint32_t kCorrelationId = 10u;
+const int32_t kMenuId = 5;
+}  // namespace
+
+class DeleteFileRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ protected:
+  void SetUp() OVERRIDE {
+    message_ = CreateMessage();
+    command_ = CreateCommand<DeleteFileRequest>(message_);
+    mock_app_ = CreateMockApp();
+  }
+
+  DeleteFileRequestPtr command_;
+  MessageSharedPtr message_;
+  MockAppPtr mock_app_;
+};
+
+TEST_F(DeleteFileRequestTest, Run_InvalidApp_UNSUCCESS) {
+  MockAppPtr invalid_app;
+  EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(invalid_app));
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL));
+  EXPECT_CALL(app_mngr_, get_settings()).Times(0);
+
+  command_->Run();
+}
+
+TEST_F(DeleteFileRequestTest, Run_HMILevelNone_UNSUCCESS) {
+  EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, hmi_level())
+      .WillOnce(Return(am::mobile_api::HMILevel::HMI_NONE));
+
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillOnce(ReturnRef(app_mngr_settings_));
+  const uint32_t num = 0;
+  EXPECT_CALL(app_mngr_settings_, delete_file_in_none())
+      .WillOnce(ReturnRef(num));
+  // Do not trying delete file twice
```

Comment:
@AByzhynar, it's mean that `delete_file_in_none_count` must be called only once.